crates/usvg-parser/src/lib.rs

```diff
@@ -89,6 +136,46 @@ impl std::fmt::Display for Error {
             Error::ParsingFailed(ref e) => {
                 write!(f, "SVG data parsing failed cause {}", e)
             }
+            Error::UnexpectedEndOfStream => {
+                write!(f, "unexpected end of stream")
+            }
+            Error::UnexpectedData(pos) => {
+                write!(f, "unexpected data at position {}", pos)
+            }
+            Error::InvalidValue => {
+                write!(f, "invalid value")
+            }
+            Error::InvalidIdent => {
+                write!(f, "invalid ident")
+            }
+            Error::InvalidChar(ref chars, pos) => {
+                // Vec<u8> -> Vec<String>
+                let list: Vec<String> = chars
```

Comment:
This seems wrong to me. In utf-8, some characters take up as many as 4 bytes? So this would break those characters into multiple characters, all being incorrect. Also, why `skip(1)`? Maybe add a comment for that if we keep it.

I feel like there is probably a standard function to do what we want here already, but I could be wrong.

Reply:
taken from https://github.com/linebender/svgtypes/blob/main/src/error.rs
line 64